packages/uploadthing/src/internal/next/pagerouter.ts

```diff
@@ -26,8 +26,12 @@ export const createNextPageApiHandler = <TRouter extends FileRouter>(
       return;
     }
 
+    const proto = (req.headers["x-forwarded-proto"] as string) ?? "http";
```

Comment:
are we certain this always exists?

Reply:
It does on Vercel, might be cases that slips through though 🤔 if only req.url returned the absolute url in pages/ like it does everywhere else :(

Reply:
Otherwise we could expose the callbackUrl as config option on nextjs pages as a fallback🤔